src/rocshmem_gpu.cpp

```diff
@@ -616,7 +617,21 @@ __device__ int rocshmem_n_pes() {
 __device__ int rocshmem_ctx_my_pe(rocshmem_ctx_t ctx) {
   GPU_DPRINTF("Function: rocshmem_ctx_my_pe\n");
 
-  return get_internal_ctx(ctx)->my_pe;
+  TeamInfo *tinfo = reinterpret_cast<TeamInfo *>(ctx.team_opaque);
+  int my_pe{get_internal_ctx(ctx)->my_pe};
+  int pe_start{tinfo->pe_start};
+  int stride{tinfo->stride};
+  int size{tinfo->size};
+
+  int translated_pe = (my_pe - pe_start) / stride;
```

Comment:
similar function here: difference is that it doesn't double check for my_pe<pe_start or my_pe>=size, which it should do anyway? [__device__ int translate_pe(rocshmem_ctx_t ctx, int pe) {](https://github.com/ROCm/rocSHMEM/blob/bce45ec32efa56c8bd28cfb4525a69ba681bb547/src/rocshmem_gpu.cpp#L267)